rskj-core/src/test/java/co/rsk/peg/BridgeUtilsLegacyTest.java

```diff
@@ -294,5 +299,69 @@ public void getAmountSentToAddress_for_mainnet_after_RSKIP293() {
         testGetAmountSentToAddress(activations, bridgeConstantsMainnet, valueToTransfer, true);
     }
 
+    private void getUTXOsForAddress_no_utxos_for_address_by_network(BridgeConstants bridgeConstants, Address address) {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+
+        BtcTransaction tx = new BtcTransaction(bridgeConstants.getBtcParams());
+        Assert.assertEquals(
+            Collections.emptyList(),
+            BridgeUtilsLegacy.getUTXOsForAddress(
+                activations,
+                bridgeConstants.getBtcParams(),
+                tx,
+                address
+            )
+        );
+    }
+
+    @Test
+    public void getUTXOsForAddress_no_utxos_for_address() {
+        Address btcAddress = Address.fromBase58(
+                bridgeConstantsMainnet.getBtcParams(),
+                "17VZNX1SN5NtKa8UQFxwQbFeFc3iqRYhem"
+        );
+        getUTXOsForAddress_no_utxos_for_address_by_network(bridgeConstantsRegtest, btcAddress);
+
+        btcAddress = Address.fromBase58(
+                bridgeConstantsRegtest.getBtcParams(),
+                "n3PLxDiwWqa5uH7fSbHCxS6VAjD9Y7Rwkj"
+        );
+        getUTXOsForAddress_no_utxos_for_address_by_network(bridgeConstantsRegtest, btcAddress);
+    }
 
+    private void getUTXOsForAddress_OK_by_network(BridgeConstants bridgeConstants) {
```

Comment:
Man, you are definitely right. The current tests should be changed to test any given address instead of the federation address.

So yes, I will consider each test case you mentioned above.